dlib/gui_widgets/widgets.cpp

```diff
@@ -6196,6 +6208,12 @@ namespace dlib
     )
     {
         auto_mutex M(m);
+
+        // trz
+        // reset it when we move to another image
+        auto_part_num = 0;
```

Comment:
You need to reset the position in a lot of cases.  For instance, when someone clicks on a different rectangle in the same image.

Reply:
I've added the reset on rectangle selection too. I'm not sure if there are any others, I couldn't find any by looking at the code and playing with the program.